libs/analytics/src/widget/orderLifecycleAnalytics.ts

```diff
@@ -0,0 +1,241 @@
+  return createDefaultTrackingPayload()
+}
+
+// Helper to set up an event handler with common logic
+const setupEventHandler = (
+  event: CowWidgetEvents,
+  payload: unknown,
+  getAdditionalProps: (payload: any) => Record<string, any>,
+  isPostedOrder = false,
+) => {
+  const analytics = getCowAnalytics()
+  if (!analytics) {
+    console.warn('Analytics instance not available for event:', EVENT_MAPPING[event])
+    return
+  }
+
+  const eventName = EVENT_MAPPING[event]
+  if (eventName) {
+    const commonProps = getCommonOrderProperties(payload, isPostedOrder)
+    const additionalProps = getAdditionalProps(payload)
+
+    analytics.sendEvent(eventName, {
+      ...commonProps,
+      ...additionalProps,
+    })
+  } else {
+    console.warn('No event mapping found for event:', event)
+  }
+}
+
+const handlePostedOrder = (payload: OnPostedOrderPayload) => {
+  setupEventHandler(
+    CowWidgetEvents.ON_POSTED_ORDER,
+    payload,
+    (p) => ({
+      sellTokenSymbol: safeGetString(p.inputToken, 'symbol'),
+      buyTokenSymbol: safeGetString(p.outputToken, 'symbol'),
+      orderType: p.orderType,
+      kind: p.kind,
+      receiver: p.receiver || '',
+      orderCreationHash: p.orderCreationHash || '',
+    }),
+    true,
+  )
+}
+
+const handleFulfilledOrder = (payload: OnFulfilledOrderPayload) => {
+  setupEventHandler(CowWidgetEvents.ON_FULFILLED_ORDER, payload, (p) => ({
+    executedSellAmount: safeGetString(p.order, 'executedSellAmount'),
+    executedBuyAmount: safeGetString(p.order, 'executedBuyAmount'),
+    executedFeeAmount: safeGetString(p.order, 'executedFeeAmount'),
+  }))
+}
+
+const handleCancelledOrder = (payload: OnCancelledOrderPayload) => {
+  setupEventHandler(CowWidgetEvents.ON_CANCELLED_ORDER, payload, (p) => ({
+    reason: 'cancelled',
+    transactionHash: p.transactionHash || '',
+  }))
+}
+
+const handleExpiredOrder = (payload: OnExpiredOrderPayload) => {
+  setupEventHandler(CowWidgetEvents.ON_EXPIRED_ORDER, payload, () => ({
+    reason: 'expired',
+  }))
+}
+
+/**
+ * Sets up event handlers for order lifecycle events.
+ * Since this is within the analytics library, we use the analytics
+ * instance directly rather than taking it as a parameter.
+ */
+export const setupEventHandlers = (eventEmitter: SimpleCowEventEmitter<CowWidgetEventPayloadMap, CowWidgetEvents>) => {
+  // Define event configurations
+  const eventConfigs = [
+    // Handle order submission
+    {
+      event: CowWidgetEvents.ON_POSTED_ORDER,
+      handler: handlePostedOrder,
+    },
+    // Handle order fulfillment
+    {
+      event: CowWidgetEvents.ON_FULFILLED_ORDER,
+      handler: handleFulfilledOrder,
+    },
+    // Handle order cancellation
+    {
+      event: CowWidgetEvents.ON_CANCELLED_ORDER,
+      handler: handleCancelledOrder,
+    },
+    // Handle order expiration
+    {
+      event: CowWidgetEvents.ON_EXPIRED_ORDER,
+      handler: handleExpiredOrder,
+    },
+  ] as const
```

Comment:
Last nitpick: I don't see this array depending on the input, so might as well define as a const outside the fn?
Unless for some reason we must set it up on every call.

Reply:
Fixed! Moved event configs outside the function as a constant since they don't depend on any inputs and don't need to be recreated on every fn call.